custom_model_runner/datarobot_drum/drum/common.py

```diff
@@ -127,6 +117,79 @@ def validate_config_fields(model_config, *fields):
         )
 
 
+def validate_model_metadata_hyperparameter(hyper_params: List) -> None:
+    """Validate hyperparameters section in model metadata yaml.
+
+    Parameters
+    ----------
+    hyper_params:
+        A list of hyperparameter definitions.
+
+    Raises
+    ------
+    DrumCommonException
+        Raised when validation fails.
+    """
+
+    def _validate_param_name(param_name: str):
+        allowed_char_re = "([a-zA-Z]+[a-zA-Z_]*[a-zA-Z]+|[a-zA-Z]+)$"
```

Comment:
One more issue - the following param name will be valid: `_my_name`
The regex will return `my_name`.

```suggestion
        allowed_char_re = "^([a-zA-Z]+[a-zA-Z_]*[a-zA-Z]+|[a-zA-Z]+)$"
```

Reply:
From the testing, `_my_name` is invalid?
```
In [13]: re_string = "([a-zA-Z]+[a-zA-Z_]*[a-zA-Z]+|[a-zA-Z]+)$"                                                                                                                                           

In [14]: re.match(re_string, "_my_name") is None                                                                                                                                                           
Out[14]: True
```

Reply:
Oh, OK. You use `match` which searches a match from the beginning of the string and opposed to `search`.